_posts/2024-11-23-dotfiles.md

```diff
@@ -0,0 +1,275 @@
+---
+title:  Dotfiles for Greenhorns
+classes: wide
+excerpt_separator: <!--more-->
+---
+
+If I could go back in time and tell a younger me a few things that would help me be a better
+software developer, it would probably be:
+
+* read the source, luke.
+* read papers.
+* create and maintain a dotfiles repo.
+
+The first two are probably obvious choices. Getting practice reading the code of the projects you
+use gives you insight that you can't get from reading the documentation. Reading papers like Amazon
+Dynamo fits the same bill and in that case, allows you to more fearlessly operate datastores. I
+would often avoid doing both because it seemed hard, and like anything new it often is hard until
+you've done it a bit and it becomes more natural.
+
+The third entry, a dotfiles repo, is probably a little less obvious. But I really love my dotfiles
+repo. It has been a force multiplier for me, personally. If you don't have or use one, maybe I can
+convince you why it's worth doing.
+
+# what is a dotfiles repo?
+
+A dotfiles repo is one that defines your terminal shell experience. This includes scripts to install
+and configure utilities you use frequently (e.g. [ripgrep](https://github.com/BurntSushi/ripgrep/),
+[homebrew](https://brew.sh/), [neovim](https://github.com/neovim/neovim),
+[tmux](https://github.com/tmux/tmux/wiki), [git](https://git-scm.com/), etc). It also can be a nice
+home to your shell configuration (e.g. configuring [ohmyzsh](https://ohmyz.sh/)). Dotfiles are also
+where the random bash scripts you write live. Do you install systemd or launchd agents? They can
+live here too.
+
+While these things are nice, the real power comes from the fact that all of this configuration is
+version controlled. I very frequently find that more junior engineers will simply start fresh when
```

Comment:
Perhaps just "many engineers" or junior and strike the more. I feel like this is useful for anyone not just early in career folks